File: src/main/java/axirassa/webapp/pages/user/LogoutUser.java

package axirassa.webapp.pages.user;

import org.apache.tapestry5.annotations.Secure;
import org.apache.tapestry5.ioc.annotations.Inject;

import axirassa.webapp.services.AxirassaSecurityService;

@Secure
public class LogoutUser {
	@Inject
	private AxirassaSecurityService security;


	String onActivate() {
		security.getSubject().logout();
		return "Index";
	}
}

File: src/main/java/axirassa/webapp/pages/user/LogoutUser.scala

package axirassa.webapp.pages.user

import org.apache.tapestry5.annotations.Secure
import org.apache.tapestry5.ioc.annotations.Inject

import axirassa.webapp.services.AxirassaSecurityService

@Secure
class LogoutUser {
	@Inject
	var security:  AxirassaSecurityService = _ 


	def onActivate() = {
		security.getSubject.logout()
		"Index"
	}
}
